src/app/comment-period/submit-comment-modal/submit-comment-modal.ts

```diff
@@ -0,0 +1,173 @@
+import { Component, OnInit } from '@angular/core';
```

Comment:
This file is all new. Validation logic, error handling, and comment submission handled here.